src/Widgets/HeaderBar.vala

```diff
@@ -47,38 +44,25 @@ public class Akira.Widgets.HeaderBar : Gtk.Box {
     public signal void new_window ();
 
     public HeaderBar () {
-        Object (orientation: Gtk.Orientation.VERTICAL, toggled: true);
```

Comment:
I would actually recommend not adding properties here unless they are specific to this constructor. If you ever decide to subclass this widget, these properties won't be inherited. So it's really better to put them in the `construct` block. I learned that one the hard way :grimacing:

Reply:
This part is not really clear to me. What's the difference between declaring these attributes inside the Object() and the construct()?

Reply:
Whatever is done in `construct` is always done any time an instance of this class (even a subclass) is created.

You can have different kinds of constructors like “MyClass ()”, “MyClass.with_some_property ()” and then in your `Object ()` you specify things that are specific to that constructor. 

So for example we have like `MyClass ()`. Let’s say we specified `Object (margin: 6)` in it. In the future if we subclass `MyClass ()` or we create a constructor like `MyClass.with_some_property ()` then that 6px margin we created won’t be applied. It’s specific to the `MyClass ()` constructor. But if we instead add `margin = 6` to our `construct` block, then it will always be applied.

So basically, unless that property is actually specific to that constructor, you probably want to put it in your `construct` block instead